x/crosschain/keeper/keeper_cross_chain_tx.go

```diff
@@ -157,16 +158,39 @@ func (k Keeper) CctxAllPending(c context.Context, req *types.QueryAllCctxPending
 		return nil, status.Error(codes.InvalidArgument, "invalid request")
 	}
 	ctx := sdk.UnwrapSDKContext(c)
+	if req.ChainId > goMath.MaxInt64 {
+		return nil, status.Error(codes.InvalidArgument, "invalid chain id")
+	}
+	pendingCrossChainTxList, err := k.GetActualPendingCCTX(ctx, int64(req.ChainId))
+	if err != nil {
+		return nil, err
+	}
+	return &types.QueryAllCctxPendingResponse{CrossChainTx: pendingCrossChainTxList}, nil
+}
+
+func (k Keeper) CctxAllPendingCount(c context.Context, _ *types.QueryAllCctxPendingCountRequest) (*types.QueryAllCctxPendingCountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	count := int64(0)
+	for _, chain := range common.DefaultChainsList() {
+		pendingCrossChainTxList, err := k.GetActualPendingCCTX(ctx, chain.ChainId)
+		if err != nil {
+			return nil, err
+		}
+		count += int64(len(pendingCrossChainTxList))
+	}
+	return &types.QueryAllCctxPendingCountResponse{CctxPendingCount: count}, nil
+}
+
+func (k Keeper) GetActualPendingCCTX(ctx sdk.Context, chainID int64) ([]*types.CrossChainTx, error) {
```

Comment:
What does `actual` means? We should add comment to explain. Shouldn't be called `GetPendingCCTXByChainID`?